eslint/cactbot-matches-type-references.js

```diff
@@ -0,0 +1,72 @@
+const findTriggerIdFromSiblingProperty = (node) => {
+  return node.parent.parent.properties.find((property) => property.key.name === 'id').value.value;
+};
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'prevent syntax issues from incorrect matches type references',
+      category: 'Syntax Issues',
+      recommended: true,
+      url: 'https://github.com/quisquous/cactbot/blob/main/docs/RaidbossGuide.md#trigger-elements',
+    },
+    fixable: 'code',
+    schema: [],
+  },
+
+  create: (context) => {
+    const netRegexTypeByTriggerId = {};
+    const matchesTypeToNetRegexType = {
```

Comment:
Should this just be implicit? e.g. prepend `Matches` and capitalize the first letter of the NetRegex type?